### PR TITLE
Turn project into node module

### PR DESCRIPTION
**Pull request by seriema**
This is a major PR and you might not want it directly. This changes the tests to unit tests that test `schema.js` directly without a server. It also removes everything to do with the homepage and running a server.

This is because of #11 and what I want to do with schematic-ipsum for my own projects: include it as a node module and generate sample data without using the website.

I will create a separate PR for just the website, that then uses this node module.

(I haven't registered anything with `npm` as you own this)

Do you agree with this? Separating the website from the sample generator?

The website project will be cleaner too. I've done this to my own projects ([npmalerts](https://github.com/seriema/npmalerts) vs [npmalerts-web](https://github.com/seriema/npmalerts-web)).

It also includes the Grunt PR #12.

### CODE REVIEW

**Comment by jonahkagan** [comment]
This is a really cool idea! Can you try to separate the Grunt PR from this PR so I can look at the changes individually?

**Comment by seriema** [comment]
Sure, but I don't know how to keep the history so it would be one big commit. Is that alright, or do you have any tips?

**Comment by seriema** [comment]
It was easier to just add back the Makefile and remove Grunt manually. Is this ok?

Do note that I made the tests output a lot less. It's probably helpful in debugging but for anyone forking and wanting to make sure the tests are working it was too chatty to be clear.

**Comment by seriema** [comment]
I've created a website version in my branch `website`. It doesn't make sense to create a PR from it so I'm just linking it here: https://github.com/seriema/schematic-ipsum/tree/website

I've left the tests in there even though they're now duplicates, but they test through the server so it kinda makes sense.

File: src/schema.coffee [hunk 1]
```diff
@@ -146,7 +146,9 @@ gen =
   # Generate a JSON object that matches the given schema filled with ipsum
   # text.
   ipsum: (schema, done) ->
-    if schema.enum?
+    if !schema?
+    then done "Needs schema"
+    else if schema.enum?
     then gen.byEnum schema, done
     else gen.byType schema, done
```
Comment (jonahkagan) [inline src/schema.coffee:153]: this might look better as a `switch` now that there are three cases

File: test/api.coffee [hunk 2]
```diff
@@ -1,24 +1,22 @@
...
 
-client = t.createJsonClient port: 3000
-
 validate = (schema, data) ->
   report = env.validate data, schema
   if _.isEmpty report.errors
     null
   else
-    console.error report.errors
+#    console.error report.errors
```
Comment (jonahkagan) [inline test/api.coffee:13]: might as well just remove these lines entirely

**Comment by jonahkagan** [comment]
:thumbsup: for bringing back the Makefile in this PR.

One thing I'm concerned about is that this removes all the documentation from this repo.

File: src/schema.coffee (hunk 1 above)
Comment (seriema) [inline src/schema.coffee:153]: It wouldn't work as a `switch`? It first checks `schema` and then `schema.enum`.

File: test/api.coffee (hunk 2 above)
Comment (seriema) [inline test/api.coffee:13]: Sure.

**Comment by seriema** [comment]
Hmm good point. I'll have to redo the PR. Hopefully I can do that this weekend.

**Pull request closed without merging**